drift-client/src/test/java/com/facebook/drift/client/TestDriftMethodInvocation.java

```diff
@@ -706,6 +710,9 @@ private static void testPropagateCancel(int expectedRetries, boolean interrupt)
             AddressSelector<?> addressSelector,
             Ticker ticker)
     {
+        ExecutorService coreExecutor = newCachedThreadPool(daemonThreadsNamed("test-retry-service-%s"));
```

Comment:
Ditto. It will create a thread leak. Please initialize it once per test class, and tear it down at `@AfterClass`